README.md

```diff
@@ -436,6 +436,33 @@ You must provide one of `--image-file` or `--image-name`.
   `LSR_QEMU_PRETTY`.
 * `--profile` - show the profile_tasks information.  The default value is `true`.
   The corresponding environment variable is `LSR_QEMU_PROFILE`.
+* `--profile-task-limit` - if using `--profile`, this specifies how many tasks
+  to show in the output.  The default is `30`.  The corresponding environment
+  variable is `LSR_QEMU_PROFILE_TASK_LIMIT`.
+* `--use-yum-cache` - Create 1 GB files in your `cache` directory for the purpose
+  of storing package cache and metadata information for the VM.  The files will
+  be named `$PLATFORM_yum_cache` and `$PLATFORM_yum_varlib`.  These are mounted
+  into the VM at `/var/cache/dnf` and `/var/lib/dnf` (or `yum` for YUM platforms).
+  If you plan to run multiple tests on the same platform, this can speed up
+  subsequent runs by installing packages from the cache rather than the network.
+  The default is `false`.  The corresponding environment variable is
+  `LSR_QEMU_USE_YUM_CACHE`.
+* `--use-snapshot` - Create a snapshot of the image using the original image as
+  the backing file.  This is useful when you want to pre-load a image for testing
+  multiple test runs, but do not want to alter the original downloaded image e.g.
+  pre-configuring package repos, pre-installing packages, etc.  This will create
+  a file called `$IMAGE_PATH.snap` e.g. `~/.cache/linux-system-roles/fedora-34.qcow2.snap`.
+  The default is `false`.  The corresponding environment variable is
+  `LSR_QEMU_USE_SNAPSHOT`.
+* `--wait-on-qemu` - This tells the script to wait for qemu to fully exit after
+  each call to `ansible-playbook`.  This can solve race conditions if you are using
+  `runqemu` to run multiple tests sequentially.  The default value is `false`.  Do
+  not use this if you are using `runqemu` interactively, especially with `--debug`.
+  This is intended only for use by automated test applications.  The corresponding
+  environment variable is `LSR_QEMU_WAIT_ON_QEMU`.
```

Comment:
**This can solve race conditions if you are using `runqemu` to run multiple tests sequentially.**

A bit confused by this sentence... Lots of tests per role are safe to run sequentially. I learned `storage` and `vpn` are problematic to do so; but others run just fine. With this observation, can we consider the above sentence means `--wait-on-qemu` is required for `storage` and `vpn` and no need for the rest of the roles?

Reply:
What I mean is that if you do
```
TEST_SUBJECTS=image.qcow2 ansible-playbook -i standard-inventory-qcow2 playbook1.yml
TEST_SUBJECTS=image.qcow2 ansible-playbook -i standard-inventory-qcow2 playbook2.yml
```
There can be a race condition where the qemu process using image.qcow2 is not fully shutdown when the second playbook run starts.  By using `--wait-for-qemu` the script will try to ensure that the qemu is fully shutdown in the first playbook run, so there is no chance that it is still running with the second playbook run.
I don't think we have seen this very often with the way we run tests currently, but now with `--use-snapshot` and `--use-yum-cache`, it is much more problematic.

Reply:
Sorry for asking this novice level question... :p
If we run these two command lines from separate window manually, we see two processes, which run independently. And we could ssh to each separagely. So, there is no race there?
```
TEST_SUBJECTS=image.qcow2 ansible-playbook -i standard-inventory-qcow2 playbook1.yml
TEST_SUBJECTS=image.qcow2 ansible-playbook -i standard-inventory-qcow2 playbook2.yml
```
Using `--use-snapshot` and `--use-yum-cache`, it's more problematic since it modifies the snapshot image file which is shared among the two tests in this example?

Reply:
> Sorry for asking this novice level question... :p If we run these two command lines from separate window manually, we see two processes, which run independently. And we could ssh to each separagely. So, there is no race there?
> 
> ```
> TEST_SUBJECTS=image.qcow2 ansible-playbook -i standard-inventory-qcow2 playbook1.yml
> TEST_SUBJECTS=image.qcow2 ansible-playbook -i standard-inventory-qcow2 playbook2.yml
> ```
> 
> Using `--use-snapshot` and `--use-yum-cache`, it's more problematic since it modifies the snapshot image file which is shared among the two tests in this example?

Hmm - ok - then perhaps it is only when using `--use-snapshot` and/or `--use-yum-cache` that causes the problem, because there is writable data shared between instances of qemu.

I have seen various failure cases:
* qemu issues an error if the yum_cache or yum_varlib file is in use by another qemu process
* qemu issues an error if the .snap file is open for writing
* the guest image will crash with a kernel panic if the .snap file is open for writing

So perhaps we only need `--wait-for-qemu` if using `--use-snapshot` and/or `--use-yum-cache`

Reply:
Thank you for the clarification!!